R/utils.R

```diff
@@ -129,31 +129,65 @@ render_dependencies <- function(dependencies, local = TRUE) {
     } else {
       "file"
     }
-    htmltools::renderDependencies(list(dep), src, encodeFunc = identity)
+    
+    # According to Dash convention, label react and react-dom as originating
+    # in dash_renderer package, even though all three are currently served
+    # u p from the DashR package
+    if (dep$name %in% c("react", "react-dom")) {
+      dep$name <- "dash-renderer"
+    }
+    
+    # The following lines inject _dash-component-suites into the src tags,
+    # as this is the current Dash convention. The dependency paths cannot
+    # be set solely at component library generation time, since hosted
+    # applications should have the app name injected as well.
+    #
+    # This is essentially analogous to this codeblock on the Python side:
+    # https://github.com/plotly/dash/blob/1249ffbd051bfb5fdbe439612cbec7fa8fff5ab5/dash/dash.py#L207
+    #
+    # Use the system file modification timestamp for the current
+    # package and add the version number of the package as a query
+    # parameter for cache busting
+    
+    if (!is.null(dep$package)) {
+      # the gsub line is to remove stray duplicate slashes, to
+      # permit exact string matching on pathnames
+      dep_path <- gsub("//+",
+                       "/",
+                       paste(dep$src$file,
+                             dep$script,
+                             sep = "/")
+      )
+      
+      full_path <- system.file(file.path(dep_path),
+                               package = dep$package)
+      
+      modified <- as.integer(file.mtime(full_path))
+    } else {
+      modified <- as.integer(Sys.time())
+    }
+    
+    # we don't want to serve the JavaScript source maps here
```

Comment:
maybe add a comment like "until we have full support for a debug=TRUE mode"?

Reply:
Good call. I could open an issue for this as well, I'll have a look at what's needed to approach parity based on what's currently available in Dash for Python.

Reply:
fixed in b6af99b6e58ce65791ff471c63b43125632d9cdd